pkg/prometheus/statefulset.go

```diff
@@ -86,6 +86,12 @@ func makeStatefulSet(p v1alpha1.Prometheus, old *v1beta1.StatefulSet, config *Co
 		},
 		Spec: *spec,
 	}
+
+	if p.Spec.ImagePullSecrets != nil && len(p.Spec.ImagePullSecrets) > 0 {
+		statefulset.Spec.Template.Spec.ImagePullSecrets = make([]v1.LocalObjectReference, len(p.Spec.ImagePullSecrets))
```

Comment:
I figured it would be safer not to keep a reference the the cache memory, but hey, you're the boss ;)